### DISCUSSION TITLE
DeltaV compatibility

### DISCUSSION THREAD

**Discussion by rkytopen** [FreeOpcUa/python-opcua#1479] [Q&A]
Set up a server that works beautifully communicating with UaExpert, but mysteriously fails to connect to a DeltaV client with no error messages but "No Comms."

Does anyone have experience with this platform?

**Comment by swamper123** [comment]
Hi,

- This library is deprecated, please try opcua-asyncio like mentioned in #1475 
- What do you mean "No Comms."? Have you got any screenshots/tracebacks from that kind of error?

**Reply by rkytopen** [reply]
Swamper-
Thanks for the suggestion. I will look into updating python on the machine and getting the newer version. In the meantime, would you believe this to be the cause of this particular issue? Basically the DeltaV client is throwing a "BAD" status as it tries to connect. Additionally there is no output from the Server indicating anything amiss. Could this have something to do with the DeltaV client certificate not being trusted by the opcua server?
<img width="774" alt="OPCUA_DV" src="https://user-images.githubusercontent.com/63860202/181633075-f2f32156-ae16-4173-9cbd-6b818ae2e03f.png">

**Reply by swamper123** [reply]
Well many things happened in the last two years in opcua-asyncio. Many things have been changed or fixed there.
I don't have a DeltaV client installed, so I can't judge about this.

Normally a Status Code "Bad" means "the value is bad but no specific reason is known". 
If a server is not trusting a certificate, it would send a "Bad_CertificateUntrusted" Status Code back. ([Here a list of possible status codes](https://reference.opcfoundation.org/v104/Core/docs/Part4/7.34.2/)).
